src/segment-loader.js

```diff
@@ -1048,12 +1049,35 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
-    this.state = 'APPENDING';
-
     const segmentInfo = this.pendingSegment_;
     const segment = segmentInfo.segment;
+    const timingInfo = this.syncController_.probeSegmentInfo(segmentInfo);
+
+    // Since video timing takes priority over audio, when we have our first timing info,
+    // determine whether this loader is dealing with video. Although we're maintaining
+    // extra state, it helps to preserve the separation of segment loader from the actual
+    // source buffers.
+    if (typeof this.isUsingVideo_ === 'undefined' && timingInfo) {
+      this.isUsingVideo_ = timingInfo.source === 'video';
+    }
+
+    // Although we support starting on audio only, and continuing playback from there, we
+    // don't support switching from audio and video to audio only. If the codecs were in
+    // the master manifest, then the renditions incompatible with the first rendition
+    // would already be blacklisted.
+    if (this.isUsingVideo_ &&
```

Comment:
WRT the other discussion on going from audio only to video+audio

If we decide that we want to support that, this will NOT trigger an error in the event that you start audio only, switch to video+audio, and then try to switch back to audio only since `this.isUsingVideo_` will be set to `false` from the initial audio only timing source